src/api/schema/topology.rs

```diff
@@ -29,44 +32,232 @@ pub struct SourceData {
 #[derive(Clone)]
 pub struct Source(SourceData);
 
-lazy_static! {
-    static ref SOURCES: Arc<RwLock<Vec<Source>>> = Arc::new(RwLock::new(vec![]));
-}
-
 #[Object]
 impl Source {
     /// Source name
     async fn name(&self) -> String {
         self.0.name.clone()
     }
 
-    /// The output type given by the source
+    /// Source output type
     async fn output_type(&self) -> SourceOutputType {
         self.0.output_type.into()
     }
+
+    /// Transform outputs
+    async fn transforms(&self) -> Vec<Transform> {
+        filter_topology(|(_name, topology)| match topology {
+            Topology::Transform(t) if t.0.inputs.contains(&self.0.name) => Some(t.clone()),
+            _ => None,
+        })
+    }
+
+    //noinspection DuplicatedCode
```

Comment:
I don't think this is required.

Reply:
Thanks, wasn’t sure how we felt about lint-disabling comments.

These are IDEA specific so they’re probably not very useful to everyone. Will nix.